xarray/backends/common.py

```diff
@@ -178,24 +206,21 @@ def add(self, source, target):
             self.sources.append(source)
             self.targets.append(target)
         else:
-            try:
-                target[...] = source
-            except TypeError:
-                # workaround for GH: scipy/scipy#6880
-                target[:] = source
+            target[...] = source
 
     def sync(self):
         if self.sources:
             import dask.array as da
+            print('self.lock == ', self.lock)
```

Comment:
Now we pass `dask.distributed.Lock(filename)` here. Interesting thing is that the test distributed test cases are failing when `da.store(..., lock=dask.distributed.Lock(filename))` is called with a 

```
TypeError: can't pickle _thread.lock objects
```

So I'm clearly missing another lock. Honestly, I'm creeping pretty far outside my comfort zone on whether or not I know what I'm doing so thoughts/suggestions at this point would be very welcome.

Reply:
This is an oversight on the Dask side.  Historically people have explicitly created locks within their functions rather than passed them around.  I believe that this is easily resolved in https://github.com/dask/distributed/pull/1738

However now I get different errors like the following in the XArray tests

```python
distributed.worker - WARNING -  Compute Failed
Function:  store_chunk
args:      (array([[ 0.02069011, -1.15082717, -0.04842995],
       [ 0.54649439,  0.39982083,  1.22082541],
       [-0.8788512 , -1.49633607, -0.49022316],
       [-1.16940102, -0.03950083,  1.15521853]]), <xarray.backends.netCDF4_.NetCDF4ArrayWrapper object at 0x7f14e7e4ab38>, (slice(0, 4, None), slice(0, 3, None)), <distributed.lock.Lock object at 0x7f14e7e7e0f0>, None, False)
kwargs:    {}
Exception: KeyError('var2',)
```

Reply:
Thanks @mrocklin - This looks like a problem I can handle.